pkg/migration/controllers/migration.go

```diff
@@ -908,15 +908,14 @@ func (m *MigrationController) runPreExecRule(migration *stork_api.Migration, mig
 		}
 	}
 	terminationChannels := make([]chan bool, 0)
-	for _, ns := range migrationNamespaces {
-		r, err := storkops.Instance().GetRule(migration.Spec.PreExecRule, ns)
-		if err != nil {
-			for _, channel := range terminationChannels {
-				channel <- true
-			}
-			return nil, err
+	r, err := storkops.Instance().GetRule(migration.Spec.PreExecRule, migration.Namespace)
+	if err != nil {
+		for _, channel := range terminationChannels {
+			channel <- true
```

Comment:
Overall I am having a little hard time understanding the implementation of terminationChannels. I do not think it is implemented correctly. But just from the point of view of this PR, terminationChannels is initialized right above this call and we will never be entering inside the for loop. 
We can simply return from this function if err is not nil just like we are doing in case of postExecRule.

Reply:
Ya, I also was not clear on that part but did not want to change the existing behaviour

Reply:
Let's remove the for loop and just return err.

Reply:
In earlier code, both GetRule and ExecuteRule were inside the loop. It made sense to append the terminationchannels and looping over inside it. Now that GetRule is outside, we can simply return as @pp511 is pointing.